### PR TITLE
Add TL-WA801ND v4 (ar71xx) and TL-WA801ND v5 (ramips)

### PR DESCRIPTION
**Pull request by roms2000**
Add support for TL-WA801ND v4 (ar71xx) (same as TL-WA801ND v3 but HWID differ)
Add support for TL-WA801ND v5 (ramips) (based on TL-WR840N v4)

Signed-off-by: MARIADASSOU Romain <roms2000@free.fr>

### CODE REVIEW

**Comment by juppin** [comment]
And you should splitt the two devices in two different commits...

File: target/linux/ar71xx/files/arch/mips/ath79/mach-tl-wa801nd-v3.c [hunk 1]
```diff
@@ -134,3 +135,19 @@ static void __init tl_wa801n_v3_setup(void)
 
 MIPS_MACHINE(ATH79_MACH_TL_WA801ND_V3, "TL-WA801ND-v3", "TP-LINK TL-WA801ND v3",
 	     tl_wa801n_v3_setup);
+
+static void __init tl_wa801n_v4_setup(void)
+{
```
[the remark this answers is not in the document]
Comment (roms2000) [reply target/linux/ar71xx/files/arch/mips/ath79/mach-tl-wa801nd-v3.c]: I did see this on others models where for some identical devices they wrote exactly the same function.
But yes, the same function can be used.

File: target/linux/ar71xx/image/tiny-tp-link.mk [hunk 2]
```diff
@@ -203,6 +203,15 @@ define Device/tl-wa801nd-v3
 endef
 TARGET_DEVICES += tl-wa801nd-v3
 
+define Device/tl-wa801nd-v4
+  $(Device/tplink-4mlzma)
```
[the remark this answers is not in the document]
Comment (roms2000) [reply target/linux/ar71xx/image/tiny-tp-link.mk]: Ok I will do.

**Comment by roms2000** [comment]
So, I will split the 2 devices and modify according with your comment.
Is there some other mistake/rules to conform to with patch ?

**Comment by juppin** [comment]
The pr template had a link to:
https://openwrt.org/submitting-patches
Here you can read about the "rules" for adding a new device...
For install procedure you could also provide a link to install instruction with the same procedure.

**Comment by bobafetthotmail** [comment]
> For install procedure you could also provide a link to install instruction with the same procedure.

Please don't do this, write install instruction in the commit message, linking to outside articles is not liked by OpenWrt core developers.

**Comment by juppin** [comment]
> Please don't do this, write install instruction in the commit message, linking to outside articles is not liked by OpenWrt core developers.

Ok. Lesson learned :-)

File: target/linux/ar71xx/image/tiny-tp-link.mk (hunk 2 above)
[the remark this answers is not in the document]
Comment (roms2000) [reply target/linux/ar71xx/image/tiny-tp-link.mk]: Thanks for your help, a new PR has been sent for tl-wa801nd-v4 only.

**Comment by roms2000** [comment]
2 PR sent, hope they will be OK.

**Comment by roms2000** [comment]
For history : 
Pull Request for TP-Link TL-WA801ND v4 (ar71xx) : https://github.com/openwrt/openwrt/pull/1111
Pull Request for TP-Link TL-WA801ND v5 (ramips) : https://github.com/openwrt/openwrt/pull/1112

**Pull request closed without merging**